File: bench/sieve.c
#include <stdio.h>
#include <stdlib.h>
#define LIMIT 100000000
int main(void) {
    unsigned char* composite = calloc(LIMIT, 1);
    long long count = 0;
    for (long long i = 2; i < LIMIT; i++) {
        if (!composite[i]) {
            count++;
            for (long long j = i * i; j < LIMIT; j += i) composite[j] = 1;
        }
    }
    printf("%lld\n", count);
    free(composite);
    return 0;
}

File: bench/sieve.py
LIMIT = 100000000

def main():
    composite = bytearray(LIMIT)
    count = 0
    for i in range(2, LIMIT):
        if not composite[i]:
            count += 1
            composite[i * i::i] = b"\x01" * len(range(i * i, LIMIT, i))
    print(count)

main()
